src/backup_restore.cpp

```diff
@@ -157,6 +157,17 @@ void BackupAndRestore::backupAndRestoreIpzVpd(types::IPZVpdMap& io_srcVpdMap,
         return;
     }
 
+    const std::string l_srcFruPath =
+        jsonUtility::getFruPathFromJson(m_sysCfgJsonObj, i_srcPath);
```

Comment:
What will be the flow if these APIs throws? Should caller handle those or this API should handle that and return to the caller or should the json utility handle that and return empty here?

Reply:
if backupAndRestoreIpzVpd is completed without exception, at the caller we declaring Backup&Restore is success and also returning the update src & dst vpd map. 
Caller is caching the exceptions from backupAndRestore* API's.

I have created the issue (https://github.com/ibm-openbmc/openpower-vpd-parser/issues/381) to relook at exceptions handling in backup&restore API's.

Reply:
ok, on the basis of the issue, I am marking this comment as closed.
Need to revisit to optimize the exception handling in this class.